sharding/indexHashedNodesCoordinator.go

```diff
@@ -591,6 +628,44 @@ func (ihgs *indexHashedNodesCoordinator) GetConsensusWhitelistedNodes(
 	return shardEligible, nil
 }
 
+// UpdatePeersListAndIndex will update the list and the index for all peers
+func (ihgs *indexHashedNodesCoordinator) UpdatePeersListAndIndex() error {
+	nodesConfig, ok := ihgs.nodesConfig[ihgs.currentEpoch]
```

Comment:
you are missing lock/unlock for relevant mutexes

Reply:
:+1:

Reply:
added